### DISCUSSION TITLE
Maximum number of 'initial_prompt' characters/tokens in transcribe()?

### DISCUSSION THREAD

**Discussion by rmohl** [openai/whisper#1386] [Q&A] [answered]
Hello!

I was wondering what might the maximum number of characters/tokens that can be given in the `initial_prompt` when using transcribe() be before they stop affecting the transcription?

Just a note: I forked the code and altered it to fix the issue described in this [comment](https://github.com/openai/whisper/discussions/117#discussioncomment-5350408). Essentially the initial prompts now affect the entire transcription, rather than just the first part of it.

In the comment refered to above, the poster says:

> Even if I pass in `initial_prompt`, it gets concatenated with the previous window transcriptions which get truncated to the model's `model.dims.n_text_ctx // 2`. For the tiny model, that's the last 223 tokens only

Would this value maybe be the max number of tokens I could use for `initial_prompt`? Or is it more complex than that?

Thanks!

**Comment by SeeknnDestroy** [comment]
hi @rmohl could you find your answer? would you mind if you share your thoughts on your experiments otherwise?

**Reply by rmohl** [reply, marked as answer]
I ended up just assuming that the max number of tokens you can use for  `initial_prompt` is 223. According to this [OpenAI website](https://platform.openai.com/tokenizer) approximately every 4 characters of text is equivalent to a single token. So the max prompt length (for my altered version of Whisper at least) is about 4 * 223 = 892 characters.

**Reply by SeeknnDestroy** [reply]
thanks :pray:

**Reply by JenuelDev** [reply]
by the way what does the prompt do?
